pallets/nfts/src/features/create_delete_item.rs

```diff
@@ -68,20 +68,20 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 			collection_details.items.saturating_inc();
 
-			AccountBalance::<T, I>::mutate(collection, &mint_to, |balance| {
-				balance.saturating_inc();
-			});
-
 			let collection_config = Self::get_collection_config(&collection)?;
 			let deposit_amount =
 				match collection_config.is_setting_enabled(CollectionSetting::DepositRequired) {
 					true => T::ItemDeposit::get(),
 					false => Zero::zero(),
 				};
-			let deposit_account = match maybe_depositor {
-				None => collection_details.owner.clone(),
-				Some(depositor) => depositor,
-			};
+			let deposit_account =
+				maybe_depositor.unwrap_or_else(|| collection_details.owner.clone());
+
+			Self::increment_account_balance(
+				collection,
+				&mint_to,
+				(&deposit_account, deposit_amount),
```

Comment:
deposit_amount is T::ItemDeposit here? Why are we not collecting the deposit for AccountBalance as well?